server/dex/dex.go

```diff
@@ -664,6 +704,33 @@ func NewDEX(ctx context.Context, cfg *DexConf) (*DEX, error) {
 	return dexMgr, nil
 }
 
+// hdKeyIndexer translates between db.FeeKeyIndexer and asset.HDKeyIndexer.
+type hdKeyIndexer struct {
+	db.FeeKeyIndexer
+}
+
+// KeyIndex gets the current index for the given extended public key, or if it
+// is unknown, creates a new entry for the key and returns the initial index.
+func (ki *hdKeyIndexer) KeyIndex(xpub string) (uint32, error) {
+	idx, err := ki.FeeKeyIndex(xpub)
+	if err == nil {
+		log.Infof("Resuming fee keys at index %d for extended pubkey %.40s...", idx, xpub)
+		return idx, nil
+	}
+	if !db.IsErrUnknownFeeKey(err) {
+		return 0, err
+	}
+	log.Infof("Creating new fee key entry for extended pubkey %.40s...", xpub)
+	return ki.CreateFeeKeyEntryFromPubKey(xpub)
```

Comment:
Why not handle this right in `(*Archiver).FeeKeyIndex`? You could seemingly preclude the need for `hdKeyIndexer` altogether if you fiddled with the method names for the interfaces.